extensions/EditComments/Extension.pm

```diff
@@ -98,13 +98,35 @@ sub page_before_template {
 
 BEGIN {
   no warnings 'redefine';
-  *Bugzilla::Comment::activity   = \&_get_activity;
-  *Bugzilla::Comment::edit_count = \&_edit_count;
+  *Bugzilla::Comment::edit_count          = \&_comment_edit_count;
+  *Bugzilla::Comment::is_editable_by      = \&_comment_is_editable_by;
+  *Bugzilla::Comment::activity            = \&_comment_get_activity;
+  *Bugzilla::User::can_edit_comments      = \&_user_can_edit_comments;
+  *Bugzilla::User::is_edit_comments_admin = \&_user_is_edit_comments_admin;
 }
 
-sub _edit_count { return $_[0]->{'edit_count'}; }
+sub _comment_edit_count { return $_[0]->{'edit_count'}; }
 
-sub _get_activity {
+sub _comment_is_editable_by {
+  my ($self, $user) = @_;
+  # Note: Does not verify that the bug is visible or editable by the user; the calling
+  # code needs to perform this validation at the bug level.
+
+  # Need to be able to edit comments via group membership
+  return 0 unless $user->can_edit_comments;
+
+  # Can always edit your own comments
+  return 1 if $self->author->id == $user->id;
+
+  # Can edit comment 0 (description) on any bug, if enabled
+  return 1 if Bugzilla->params->{allow_global_initial_comment_editing}
+    && $self->count == 0;
+
+  # Otherwise not editable
```

Comment:
Need to add case where comment is editable by admins.

return 1 if $user->in_group(Bugzilla->params->{edit_comments_admins_group});